[Darshan P Daryanani Dave] iP

darshan152 opened the pull request:
# Duke
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Features
- Add Task
- Remove Task
- Mark Task as Done
- Mark Task as Undone
- Add Times
- Search for Tasks

Task Types:
1. **ToDo**
2. _Events_
3. **_Deadlines_**

Extentions:
- [ ] Reminders ⌚ 
- [ ] GUI

Contains classses like `Main`, `UI`, `Storage`
```Java
public class Main {
    public static void main(String[] args) {
        ....
    }
}
```

Commit:
Add Gradle support

darshan152 committed:
Level-1

darshan152 committed:
Level-2

darshan152 committed:
Level-3

darshan152 committed:
Level-4

darshan152 committed:
A-TextUiTesting

darshan152 committed:
Handling Errors

darshan152 committed:
Delete Functionality

darshan152 committed:
Adding Test Directory

darshan152 committed:
Move parsing of input to Duke

Responsibility of parsing input moved from TaskList to Duke

This means that TaskList would now not need to deal with raw input.
This was done in order to simplify the job of TaskList

darshan152 committed:
Add saving tasks to csv

Add functionality to save tasks to a csv file in the data folder.

This would help to add functionality to save tasks over different runs of the
program

darshan152 committed:
Add reading tasks from csv

Adding function to read data from csv in order to keep data across different runs of the program

darshan152 committed:
Complete save implementation

Users will now be able to save tasks across different runs of app

darshan152 committed:
Add date functionality in events and deadlines

Tasks are able to recognize dates and print them as dates.

darshan152 committed:
Add datetime functionality

Added ability to store datetime in Event and Deadline

darshan152 committed:
Merge pull request #1 from darshan152/branch-Level-7

Branch level 7

darshan152 committed:
Merge branch 'branch-Level-8'

darshan152 committed:
Add parser class

Refactor code to add a parser class in order to conform to OOP

darshan152 committed:
Improve parser class

Removed any parsing of inputs done by Duke and moved to Parser class

darshan152 committed:
Add ui class

Add Ui class in order to conform better to OOP. Ui class handles interactions with user

darshan152 committed:
Add ui functionality

Adding more responsibility for the Ui class

darshan152 committed:
Add package

Added files into package duke

darshan152 committed:
Add junit tests for tasks

Add Junit test to ensure tasks are created as intended

darshan152 committed:
Add jar file

Added Jar file for users to easily run application

darshan152 committed:
Add javadoc comments

Adding JavaDoc comments to improve understanding of code

darshan152 committed:
Refactor to conform to coding standard

Refactoring code to conform to coding standards given

darshan152 committed:
Add find feature

Adding a find feature so users would be able to easily find their tasks

darshan152 committed:
Merge pull request #3 from darshan152/Branch-A-JavaDoc

Add javadoc comments

darshan152 committed:
Merge branch 'master' into A-CodingStandard

darshan152 committed:
Merge pull request #4 from darshan152/A-CodingStandard

Refactor to conform to coding standard

darshan152 committed:
Merge branch 'master' into Branch-Level-9

darshan152 committed:
Merge pull request #5 from darshan152/Branch-Level-9

Add find feature

darshan152 committed:
Merge pull request #6 from darshan152/add-gradle-support

Add Gradle support

darshan152 committed:
Add gradle

Automate project build using Gradle

darshan152 committed:
Merge pull request #7 from darshan152/A-Gradle

A-gradle

darshan152 committed:
Add support for unit tests

Add support for gradle to run unit test

darshan152 committed:
Merge pull request #8 from darshan152/Branch-A-Gradle

Add support for unit tests

darshan152 committed:
Add gui for duke

Add GUI for duke that echos users message.

darshan152 committed:
Add functionality to the gui

Add functionality to the GUI so that it works like a chatbot

darshan152 committed:
Fix checkstyle

yunbinmo commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,158 @@
+    /**
+     * Arraylist of Tasks
+     */
+    private final ArrayList<Task> actions = new ArrayList<>();
```
Perhaps using `tasks` as the variable name would make more sense.

yunbinmo commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,158 @@
+     * @return Boolean representing success of add operation
+     * @throws EmptyDescriptionException if description is empty
+     */
+    public boolean add(TaskType type, String[] inputs) throws EmptyDescriptionException {
```
Perhaps `addTask` would be a better name to describe what this method does, but if this name is clear enough in its context then it should be fine.

yunbinmo commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,60 @@
+     *
+     * @return String array with parsed inputs from user
+     */
+    public static String[] getInputs() {
```
`getUserInputs` might be a better method name.

yunbinmo commented on src/test/java/duke/DeadlineTest.java:
```diff
@@ -0,0 +1,18 @@
+
+public class DeadlineTest {
+    @Test
+    public void getDetails_newDeadlineCreated_returnStringArray() {
```
I like how you follow the alternative naming convention for testing methods here!

darshan152 committed:
Add javafx implementation

Adding JavaFX in order to make the UI easier to modify

darshan152 committed:
Remove cli and original javafx

Removed CLI implementation and original JavaFX implementations

darshan152 committed:
Rename methods

Renaming methods to follow naming conventions better and make them more intuitive

darshan152 committed:
Remove unnecessary methods

Remove unnecessary methods to clean up code

darshan152 committed:
Remove testing for ui

Remove tests for text UI as text UI was removed.

geetengtan commented on src/main/java/duke/DataStore.java:
```diff
@@ -0,0 +1,96 @@
+/**
+ * Represents Storage of the tasks
+ */
+public class DataStore {
```
Class name should be nouns

geetengtan commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,111 @@
+                break;
+            case EVENT:
+                try {
+                    wasAddSuccess = list.add(TaskType.EVENT, parsedInput);
```
The process of adding different tasks can be abstracted for better readability since it has also been repeated

geetengtan commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,158 @@
+            break;
+        case DEADLINE:
+            task = new Deadline(inputs[1], inputs[2]);
+
```
This line should be omitted

geetengtan commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,66 @@
+     * @return Command enum of command to execute
+     * @throws CommandNotFoundException when command is not found
+     */
+    public static Command parseCommand(String command) throws CommandNotFoundException {
```
Very tidy method, very good, very readable 🙂

geetengtan reviewed (commented):
Well done. Very organized code structure

darshan152 committed:
Fix issue #9

darshan152 committed:
Handle index out of bounds and number format exceptions

Handle index out of bounds and number format exceptions in Duke to make it more user friendly

darshan152 committed:
Merge pull request #10 from darshan152/Branch-Level-10

Branch level 10

darshan152 committed:
Bug fixes

Fixed issue with saving data
Added support messaged for wrong Delete commands
Added title for app window
Added welcome message
Added message for empty list
Added message for empty find
Removed warning messages for fxml files

darshan152 committed:
Update MainWindow.java

ivorcmx commented on src/main/java/duke/DataStore.java:
```diff
@@ -0,0 +1,103 @@
+/**
+ * Represents Storage of the tasks
+ */
+public class DataStore {
```
Perhaps the class name could be changed to better suit naming conventions

ivorcmx commented on src/main/java/duke/DataStore.java:
```diff
@@ -0,0 +1,103 @@
+                }
+                csvReader.close();
+            }
+
```
The empty line could maybe be removed

ivorcmx commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,171 @@
+        }
+
+        for (int i = 0; i < tasks.size(); i++) {
+
```
Perhaps you can remove the empty line for better coding standard

ivorcmx commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,144 @@
+        Boolean wasAddSuccess = null;
+
+
+        switch (actionType) {
```
Very neat way of displaying the different cases and what the code does!

ivorcmx reviewed (commented):
Overall a fantastic project with little comments to be made about it. I agree with the other reviewers comments as well. Coding standards were mostly followed.

darshan152 committed:
Update build.gradle

darshan152 committed:
Add assert statements

Adding assert statements to ensure that the application runs as expected.

darshan152 committed:
Improve code quality

Changes made to improve quality of code.
This will help to ensure that future developers will have an easier time
reading and editing the code.
Code quality standards adhered to  are found in the module textbook.

darshan152 committed:
Remove some magic literals

darshan152 committed:
Merge pull request #11 from darshan152/branch-A-CodeQuality

Merge branch-A-CodeQuality

darshan152 committed:
Create gradle.yml

darshan152 committed:
Revert "Remove some magic literals"

This reverts commit 195375f74a27176c982e3f9c4586cad2d8457524.

darshan152 committed:
Create gradle.yml

darshan152 committed:
Implemented Sort functionality

darshan152 committed:
Merge pull request #13 from darshan152/C-Sort

Implement sorting

darshan152 committed:
Add rounded corners for text box

darshan152 committed:
Separate dialog boxes for input and output

darshan152 committed:
Update to follow checkstyle

darshan152 committed:
Create Ui.png

darshan152 committed:
Merge pull request #15 from darshan152/branch-A-BetterGui

Improve UI of Duke

darshan152 committed:
Change javadoc comments

darshan152 committed:
Merge branch 'master' of https://github.com/darshan152/ip

darshan152 committed:
Update readme documentation

darshan152 committed:
Update bye functionality

darshan152 committed:
Update gui

darshan152 committed:
Update Main.java

darshan152 committed:
Update MainWindow.fxml

darshan152 committed:
Update MainWindow.fxml

darshan152 committed:
Update README.md

darshan152 committed:
Merge branch 'master' into branch-A-CodeQuality

darshan152 committed:
Merge pull request #16 from darshan152/branch-A-CodeQuality

Branch a code quality
